src/bcf/annotate_dgidb.rs

```diff
@@ -33,37 +33,47 @@ pub fn annotate_dgidb(
     vcf_path: &str,
     api_path: String,
     field_name: &str,
+    genes_per_request: usize,
 ) -> Result<(), Box<dyn Error>> {
-    let gene_drug_interactions = request_interaction_drugs(vcf_path, api_path)?;
+    let gene_drug_interactions = request_interaction_drugs(vcf_path, api_path, genes_per_request)?;
     modify_vcf_entries(vcf_path, gene_drug_interactions, field_name)
 }
 
 fn request_interaction_drugs(
     vcf_path: &str,
-    mut api_path: String,
+    api_path: String,
+    genes_per_request: usize,
 ) -> Result<Option<HashMap<String, Vec<(String, Vec<String>)>>>, Box<dyn Error>> {
     let mut genes = collect_genes(vcf_path)?;
```

Comment:
I know you didn't change this in the PR, but I was wondering if this needs to be `mut`? Is that necessary for the slicing below?